cmd/clusterctl/client/tree/discovery.go

```diff
@@ -65,10 +67,14 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 	}
 
 	// Adds control plane
-	controlPLane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
-	if err == nil {
-		tree.Add(cluster, controlPLane, ObjectMetaName("ControlPlane"), GroupingObject(true))
+	if cluster.Spec.ControlPlaneRef == nil {
+		return nil, errors.New(fmt.Sprintf("spec.controlPlaneRef for Cluster %v is nil, this is unexpected", client.ObjectKeyFromObject(cluster)))
 	}
+	controlPlane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("failed to get controlplane for cluster %v", client.ObjectKeyFromObject(cluster)))
```

Comment:
```suggestion
		return nil, errors.Wrapf(err, "failed to get ControlPlane for Cluster %v", client.ObjectKeyFromObject(cluster))
```
You can also drop the fmt.Sprintf here

Reply:
+1